src/reducers/counterValues.js

```diff
@@ -18,17 +22,19 @@ const handlers = {
   UPDATE_COUNTER_VALUES: (
     state: CounterValuesState,
     { payload: counterValues }: { payload: CounterValuesState }
-  ): CounterValuesState => merge(state, counterValues)
+  ): CounterValuesState => merge({ ...state }, counterValues)
 };
 
 const getPairHistory = state => (coinTicker, fiat) => {
   const byDate = get(state, `counterValues.${coinTicker}.${fiat}`);
-  return date => {
-    const d = date || new Date(); // TODO later, we need instead to branch the lookup to pick from latest value
-    return (byDate && byDate[formatCounterValueDay(d)]) || 0;
-  };
+  return date =>
+    (byDate && (date ? byDate[formatCounterValueDay(date)] : byDate.latest)) ||
```

Comment:
mmh, it does not fallback on `byDate.latest ` in case `byDate[formatCounterValueDay(date)]` is falsy/0. so we need to do things like `date ? byDate[formatCounterValueDay(date)] || byDate.latest : byDate.latest` , a more concise way is `date && byDate[formatCounterValueDay(date)] || byDate.latest`